overlord/snapshotstate/snapshotstate_test.go

```diff
@@ -946,11 +958,15 @@ func (snapshotSuite) TestRestoreIntegrationFails(c *check.C) {
 	tasks := change.Tasks()
 	c.Check(tasks, check.HasLen, 3)
 	for _, task := range tasks {
-		c.Check(task.Status(), check.Equals, state.ErrorStatus)
 		if strings.Contains(task.Summary(), `"too-snap"`) {
+			c.Check(task.Status(), check.Equals, state.ErrorStatus)
 			c.Check(strings.Join(task.Log(), "\n"), check.Matches, `\S+ ERROR mkdir \S+: permission denied`)
 		} else {
-			c.Check(strings.Join(task.Log(), "\n"), check.Matches, `\S+ ERROR context canceled`)
+			if task.Status() == state.ErrorStatus {
```

Comment:
A comment would be nice why sometimes we go into "ErrorStatus" and sometimes into "UndoStatus". I.e. why we handle both cases in the test.